js/src/Map.js

```diff
@@ -195,11 +197,31 @@ var LeafletMapView = utils.LeafletDOMWidgetView.extend({
                 var c = e.target.getCenter();
                 that.model.set('center', [c.lat, c.lng]);
                 that.dirty = false;
-            }
+            };
+            var style_view = Object.keys(that.model.get('default_style').views)[0];
+            that.model.get('default_style').views[style_view].then(function(view) {
+                view.style();
+            });
+            var dragging_style_view = Object.keys(that.model.get('dragging_style').views)[0];
+            that.model.get('dragging_style').views[dragging_style_view].then(function(view) {
+                view.unstyle();
+            });
             that.model.update_bounds().then(function() {
                 that.touch();
             });
         });
+
+        this.obj.on('movestart', function (e) {
+            var style_view = Object.keys(that.model.get('default_style').views)[0];
```

Comment:
Still doesn't work.. same error as before. It cannot find the view :thinking:
For testing, just after this line I wrote:
``` javascript
console.log(style_view)
```
and returned `undefined`

Reply:
Yeah I see that :( 

It looks like the `style` property must be named "style", nothing else.

Reply:
Yes, I don't know why.. and also wondering if `dragging_style` will work (I guess not)

Reply:
Maybe it is hardcoded somewhere in ipywidgets... I'll ask.

Every time we think we have a solution it does not work! Sorry for that.

Reply:
Yes, it is actually some logic in the `DomWidgetView` class (`LeafletMapView inherits from it). See [this function](https://github.com/jupyter-widgets/ipywidgets/blob/1df5b73812a4186504c120e327389a8084a2f226/packages/base/src/widget.ts#L798). They create a view for the `style` property, only the `style` property, so if it has not the name `style` no view will be created.

Either we implement the same logic for `default_style` and `dragging_style`, or we come up with a solution not using the `Style` widget.

Reply:
I though in a new solution, but (as always) I'm struggling.. 

This'd be in the python side:

``` python
style = InstanceDict(MapStyle).tag(sync=True, **widget_serialization)
dragging_style = Unicode('move').tag(sync=True, o=True)
default_style = Unicode('hand').tag(sync=True, o=True)
```
(I don't know what's the `o=True` for, but just in case...)

Then in `LeafletMapModel`:

``` javascript
...
style: null,
default_style: 'hand',
dragging_style: 'move',
...
```
I defined serializers for these two variables just in case (I'm not sure if it's needed)..

Then, in `leaflet_events`:
``` javascript
this.obj.on('movestart', function (e) {
    var style_model = that.model.get('style');
    var style_view = style_model.views[Object.keys(style_model.views)[0]]
    style_view.then(function(view){
        view.model.set('cursor', that.model.get('dragging_style'))
    })
});
this.obj.on('moveend', function (e) {
    ...
    var style_model = that.model.get('style');
    var style_view = style_model.views[Object.keys(style_model.views)[0]]
    style_view.then(function(view){
        view.model.set('cursor', that.model.get('default_style'))
    })
    ...
});
```
It works for the `movestart` event... but it doesn't for the `moveend`.. and I don't know why.. I tried to figure it out, but no luck :thinking:

Reply:
Indeed, it would be a good approach. You don't need `o=True`, this is for specifying if a trait is a Leaflet option, `dragging_style` and `default_style` are not. You also don't need deserializers for `dragging_style` and `default_style`, only for `style`.

Do you have a JavaScript error? Or it just doesn't do anything?
Out of curiosity, do you mind trying the following:
```javascript
that.model.get('style').set('cursor', that.model.get('default_style'));
```
instead of
```javascript
    var style_model = that.model.get('style');
    var style_view = style_model.views[Object.keys(style_model.views)[0]]
    style_view.then(function(view){
        view.model.set('cursor', that.model.get('default_style'))
    })
```

Reply:
I tried that before, but in the `moveend` event, and as it didn't work, I changed it.. but now I've tried to do it in the `movestart` event and works fine.. but it still does not work in the `moveend` event.. do I PR?

Reply:
Also, what do you think if I change the variable names to be more specific to `cursor`?? because in the future another style could be modified.. like: `default_cursor` instead of `default_style` and `dragging_cursor` instead of `dragging_style`..

Reply:
I don't get any JS errors.. it just doesn't behave as expected

Reply:
> do I PR?

Yes please :)

> what do you think if I change the variable names to be more specific to cursor??

I agree. Or you could make `default_style` and `dragging_style` dictionaries containing the `cursor` key with its value.